### PR TITLE
[Port] Hardsuit helmets can now be used as internals masks.

### PR DESCRIPTION
**Pull request by SynnGraffkin**
## About The Pull Request

Credit to @jakeramsay007 

Also reworks internals code a little to be much more friendly. Any head or mask item with the flag should now function as an internals mask if it has the ALLOWSINTERNALS flag.

image

## Why It's Good For The Game

It's dumb you need a mask, on top of your spaceworthy helmet, to use internals. It also works for Plasmeme helmets.

You can also smoke in your suit too if you want.

image

## Changelog
:cl: Chayse
add: Assorted space-worthy helmets can now act as masks for internals.
refactor: Internals code can now check any item with the ALLOWSINTERNALS flag through the GET_INTERNAL_SLOTS define. For now this only checks head and mask slots, since those are the most realistically speaking usable ones.
/:cl:

### CODE REVIEW

**Comment by dinnerwoerror** [comment]
dude how many times has the doctor let the mask burn up and then they're fumbling for it when im in the cloning room and asking them for the mask

File: code/game/objects/items/tanks/tanks.dm [hunk 1]
```diff
@@ -33,15 +33,22 @@
...
+			//Wasp Port Begin - Citadel Internals
+			var/obj/item/clothing/check
+			var/internals = FALSE
+
+			for(check in GET_INTERNAL_SLOTS(H))
+				if(istype(check, /obj/item/clothing/mask))
+					var/obj/item/clothing/mask/M = check
+					if(M.mask_adjusted)
+						M.adjustmask(H)
+				if(CHECK_BITFIELD(check.clothing_flags, ALLOWINTERNALS))
```
Comment (MarkSuckerberg) [inline code/game/objects/items/tanks/tanks.dm:45]:
```suggestion
				if(check.clothing_flags & ALLOWINTERNALS))
```

File: code/modules/mob/living/carbon/life.dm [hunk 2]
```diff
@@ -311,11 +311,19 @@
 	breath.temperature = bodytemperature
 
 /mob/living/carbon/proc/get_breath_from_internal(volume_needed)
+	//Wasp Port Begin - Citadel Internals
+	var/obj/item/clothing/check
+	var/internals = FALSE
+
+	for(check in GET_INTERNAL_SLOTS(src))
+		if(CHECK_BITFIELD(check.clothing_flags, ALLOWINTERNALS))
```
Comment (MarkSuckerberg) [inline code/modules/mob/living/carbon/life.dm:319]:
```suggestion
		if(check.clothing_flags & ALLOWINTERNALS))
```

File: code/_onclick/hud/screen_objects.dm [hunk 3]
```diff
@@ -300,16 +300,21 @@
...
+		//Wasp Port Begin - Citadel Internals
+			var/obj/item/clothing/check
+			var/internals = FALSE
+
+			for(check in GET_INTERNAL_SLOTS(C))
+				if(istype(check, /obj/item/clothing/mask))
+					var/obj/item/clothing/mask/M = check
+					if(M.mask_adjusted)
+						M.adjustmask(C)
+				if(CHECK_BITFIELD(check.clothing_flags, ALLOWINTERNALS))
```
Comment (MarkSuckerberg) [inline code/_onclick/hud/screen_objects.dm:312]:
```suggestion
				if(check.clothing_flags & ALLOWINTERNALS))

```

File: code/__DEFINES/flags.dm [hunk 4]
```diff
@@ -4,6 +4,16 @@
...
+//Wasp Port - Citadel Internals - BFAT "I dunno what else Bitfields do but I needed them, so if it can be done without it, send help"
+//for convenience
+#define ENABLE_BITFIELD(variable, flag) (variable |= (flag))
+#define DISABLE_BITFIELD(variable, flag) (variable &= ~(flag))
+#define CHECK_BITFIELD(variable, flag) (variable & (flag))
+#define TOGGLE_BITFIELD(variable, flag) (variable ^= (flag))
+
+#define CHECK_MULTIPLE_BITFIELDS(flagvar, flags) (((flagvar) & (flags)) == (flags))
+//Wasp Port End
+
```
Comment (MarkSuckerberg) [inline code/__DEFINES/flags.dm:16]: Shouldn't(:tm:) need this if you accept the suggestions

**Comment by MarkSuckerberg** [comment]
FECK I LEFT AN EXTRANEOUS PARENTHESIS ON ALL OF EM

**Comment by SynnGraffkin** [comment]
I fixed that

**Comment by MarkSuckerberg** [comment]
thank chroist

**Comment by tiramisuapimancer** [comment]
Smoker quirk is no longer a meme on plasmemes

**Comment by dee-dubya** [comment]
Hold on, I thought we already had this from a previous tg merge?

**Comment by SynnGraffkin** [comment]
No, I had to **unfuck** spacesuits having batteries.

**Comment by triplezeta** [comment]
no

**Comment by ADepressedPyroshark** [comment]
Does this include explorer's suit. Like can you do this with the hood?

**Pull request merged**